admin_ui/src/components/CSVButton.vue

```diff
@@ -0,0 +1,94 @@
+<template>
+    <a class="button">
+        <downloadexcel
+            :before-finish="finishDownload"
+            :before-generate="startDownload"
+            :fetch="fetchExportedRows"
+            :name="tableName + '.csv'"
+            type="csv"
+        >
+            <span>
+                <font-awesome-icon icon="file-csv" />Export as CSV
+            </span>
+        </downloadexcel>
+    </a>
+</template>
+
+<script lang="ts">
+import axios from "axios"
+import downloadexcel from "vue-json-excel"
+import * as i from "../interfaces"
+
+export default {
+    props: ["tableName"],
+    data() {
+        return {
+            tableName: this.tableName,
+        }
+    },
+    components: {
+        downloadexcel,
+    },
+    methods: {
+        // Export data as csv from json:
+        async fetchExportedRows() {
+            const params = this.$store.state.filterParams
+            const tableName = this.$store.state.currentTableName
```

Comment:
You should be able to get `tableName` from `this.tableName` instead of reading it from the store.